lib/ecto/changeset.ex

```diff
@@ -3272,10 +3311,15 @@ defmodule Ecto.Changeset do
 
   """
   def exclusion_constraint(changeset, field, opts \\ []) do
-    constraint = opts[:name] || "#{get_source(changeset)}_#{get_field_source(changeset, field)}_exclusion"
+    name = opts[:name] || "#{get_source(changeset)}_#{get_field_source(changeset, field)}_exclusion"
     message    = message(opts, "violates an exclusion constraint")
     match_type = Keyword.get(opts, :match, :exact)
-    add_constraint(changeset, :exclusion, to_string(constraint), match_type, field, message, :exclusion)
+    constraint = case name do
+      %Regex{} -> name
+      _ -> to_string(name)
+    end
```

Comment:
Let's encapsulate this case into a tiny function called `normalize_constraint/1`.

Reply:
Done!